app/code/Magento/Catalog/Block/Product/View/GalleryOptions.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Catalog\Block\Product\View;
+
+use Magento\Framework\View\Element\Block\ArgumentInterface;
+use Magento\Framework\Serialize\Serializer\Json;
+use Magento\Catalog\Block\Product\Context;
+use Magento\Framework\Stdlib\ArrayUtils;
+
+class GalleryOptions extends AbstractView implements ArgumentInterface
+{
+    /**
+     * @var Json
+     */
+    protected $jsonSerializer;
```

Comment:
All new methods & properties should not be protected. Use private instead
```suggestion
    private $jsonSerializer;
```